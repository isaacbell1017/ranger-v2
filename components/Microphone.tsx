import React, { useEffect } from 'react';
import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition';

const Microphone = () => {
  const {
    transcript,
    listening,
    finalTranscript,
    resetTranscript,
    browserSupportsSpeechRecognition,
    isMicrophoneAvailable
  } = useSpeechRecognition({ commands: true });

  const [language, useLanguage] = 'en-US'

  useEffect(() => {
    if (finalTranscript !== '') {
      console.log('transcript: ', finalTranscript);
    }
  }, [finalTranscript]);

  const listen = (currLanguage = language) => {
    SpeechRecognition.startListening({
      language: currLanguage,
      continuous: true,
    });
  };

  const stop = () => {
    SpeechRecognition.stopListening();
  };

  return (
    <>
      <h1>Listening? {listening ? 'Yes' : 'No'}</h1>
      <button onClick={() => resetTranscript()}>Reset</button>
      <button onClick={() => listen()}>Listen</button>
      <button onClick={() => stop()}>Stop</button>

      <hr />
      <p>{transcript}</p>
    </>
  );
};

export default Microphone;
