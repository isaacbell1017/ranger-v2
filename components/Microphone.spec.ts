import React from 'react';

import renderer from 'react-test-renderer';
import 'regenerator-runtime/runtime'

import Link from './Microphone';
import Microphone from './Microphone';

it('renders correctly', () => {
  const tree = renderer
    .create(<Microphone />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
