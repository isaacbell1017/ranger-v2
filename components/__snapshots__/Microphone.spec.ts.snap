// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <h1>
    Listening? 
    No
  </h1>,
  <button
    onClick={[Function]}
  >
    Reset
  </button>,
  <button
    onClick={[Function]}
  >
    Listen
  </button>,
  <button
    onClick={[Function]}
  >
    Stop
  </button>,
  <hr />,
  <p />,
]
`;
